plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/session/CodeGenerationState.kt

```diff
@@ -156,8 +158,12 @@ private suspend fun CodeGenerationState.generateCode(codeGenerationId: String, m
                     -> featureDevServiceError(message("amazonqFeatureDev.exception.prompt_refusal"))
                     codeGenerationResultState.codeGenerationStatusDetail()?.contains(
                         "EmptyPatch"
-                    ),
-                    -> featureDevServiceError(message("amazonqFeatureDev.exception.guardrails"))
+                    ) -> {
+                        if (codeGenerationResultState.codeGenerationStatusDetail()?.contains("NO_CHANGE_REQUIRED")!!) {
+                            noChangeRequiredException()
+                        }
+                        featureDevServiceError(message("amazonqFeatureDev.exception.guardrails"))
```

Comment:
Not from this change, but why EmptyPatch is using guardrails error message.

Reply:
It leads to the same message as we have on VS Code, it just have different naming.